lang/rust/rust.talon

```diff
@@ -48,6 +48,7 @@ toggle use: user.code_toggle_libraries()
 ## Simple aliases
 borrow: "&"
 borrow mutable: "&mut "
+state (async | asynchronous): "async "
```

Comment:
I use "a sink" for this, as I found recognition better.  Not sure our stance regarding these kind of games to improve recognition tho.  cc/ @knausj85 @lunixbochs

Reply:
At some point it might be worth adding this under a common tag to make this consistent between languages.

Reply:
Fwiw we support this [as a keyword](https://github.com/knausj85/knausj_talon/blob/3eab2c9525c3896718152c1f88e0f2d0ceee1237/lang/javascript/javascript.py#L48) in Javascript